x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/settings_flyout/schedule/schedules_table/index.tsx

```diff
@@ -0,0 +1,148 @@
+import type { AttackDiscoverySchedule } from '@kbn/elastic-assistant-common';
+
+import * as i18n from './translations';
+
+import { useColumns } from './use_columns';
+import { useFindAttackDiscoverySchedules } from '../logic/use_find_schedules';
+import { useEnableAttackDiscoverySchedule } from '../logic/use_enable_schedule';
+import { useDisableAttackDiscoverySchedule } from '../logic/use_disable_schedule';
+import { useDeleteAttackDiscoverySchedule } from '../logic/use_delete_schedule';
+
+const DEFAULT_PAGE_SIZE = 10;
+const DEFAULT_SORT_FIELD = 'name';
+const DEFAULT_SORT_DIRECTION = 'asc';
+
+/**
+ * Table Component for displaying Attack Discovery Schedules
+ */
+export const SchedulesTable: React.FC = React.memo(() => {
+  const [pageIndex, setPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(DEFAULT_PAGE_SIZE);
+  const [sortField, setSortField] = useState<keyof AttackDiscoverySchedule>(DEFAULT_SORT_FIELD);
+  const [sortDirection, setSortDirection] = useState<'asc' | 'desc'>(DEFAULT_SORT_DIRECTION);
+
+  const { data: { schedules, total } = { schedules: [], total: 0 }, isLoading: isDataLoading } =
+    useFindAttackDiscoverySchedules({
+      page: pageIndex,
+      perPage: pageSize,
+      sortField,
+      sortDirection,
+    });
+
+  const pagination = useMemo(() => {
+    return {
+      pageIndex,
+      pageSize,
+      totalItemCount: total,
+    };
+  }, [pageIndex, pageSize, total]);
+
+  const sorting = useMemo(() => {
+    return {
+      sort: {
+        field: sortField,
+        direction: sortDirection,
+      },
+    };
+  }, [sortDirection, sortField]);
+
+  const onTableChange = useCallback(
+    ({ page, sort }: CriteriaWithPagination<AttackDiscoverySchedule>) => {
+      if (page) {
+        setPageIndex(page.index);
+        setPageSize(page.size);
+      }
+      if (sort) {
+        const { field, direction } = sort;
+        setSortField(field);
+        setSortDirection(direction);
+      }
+    },
+    []
+  );
+
+  const [isTableLoading, setTableLoading] = useState(false);
+
+  const isSchedulesLoading = isDataLoading || isTableLoading;
+
+  const { mutateAsync: enableAttackDiscoverySchedule } = useEnableAttackDiscoverySchedule();
+  const { mutateAsync: disableAttackDiscoverySchedule } = useDisableAttackDiscoverySchedule();
+  const { mutateAsync: deleteAttackDiscoverySchedule } = useDeleteAttackDiscoverySchedule();
+
+  const getScheduleData = useCallback(
+    (scheduleId: string) => {
+      if (!isSchedulesLoading && schedules.length) {
+        return schedules.find((item) => item.id === scheduleId);
+      }
+    },
+    [isSchedulesLoading, schedules]
+  );
+
+  const openScheduleDetails = useCallback((scheduleId: string) => {
+    // TODO: implement attack discovery schedule details
+  }, []);
+  const enableSchedule = useCallback(
+    async (id: string) => {
+      setTableLoading(true);
+      await enableAttackDiscoverySchedule({ id });
+      setTableLoading(false);
```

Comment:
Consider a `try` / `finally` here to `setTableLoading(false)` if the `await` fails here, and throughout the PR where there's a similar pattern

Reply:
done in https://github.com/elastic/kibana/pull/217917/commits/38585fbef19ecfdf1ef2764d67edeead4f00329e